ballerina-tests/tests/test.bal

```diff
@@ -4,6 +4,198 @@ import ballerina/lang.value;
+    final http:Client clientEndpoint = check new ("http://localhost:3000");
+    string jsonFilePath = "./tests/resources/httpHeaderTest.json";
+    string readString = check io:fileReadString(jsonFilePath);
+    string replacedString = regex:replaceAll(readString, "(FUNC_NAME)", "httpHeaderTest7");
+    json readJson = check value:fromJsonString(replacedString);
+    json resp = check clientEndpoint->post("/post-hello-httpHeaderTest7", readJson);
+    json expectedResp = {"Outputs":{"resp":{"statusCode":"201", "headers":{"Content-Type":"text/plain"}, "body":"az-func-http-test.azurewebsites.net"}}, "Logs":[], "ReturnValue":null};
+    test:assertEquals(resp, expectedResp);
+}
+
+@test:Config {}
+function httpHeaderTest8() returns error? {
+    final http:Client clientEndpoint = check new ("http://localhost:3000");
+    string jsonFilePath = "./tests/resources/httpHeaderTest.json";
+    string readString = check io:fileReadString(jsonFilePath);
+    string replacedString = regex:replaceAll(readString, "(FUNC_NAME)", "httpHeaderTest8");
+    json readJson = check value:fromJsonString(replacedString);
+    json resp = check clientEndpoint->post("/post-hello-httpHeaderTest8", readJson);
+    json expectedResp = {"Outputs":{"resp":{"statusCode":400, "body":"no header value found for 'Hoste'", "headers":{"Content-Type":"text/plain"}}}, "Logs":[], "ReturnValue":null};
+    test:assertEquals(resp, expectedResp);
+}
+
+@test:Config {}
+function httpHeaderTest9() returns error? {
+    final http:Client clientEndpoint = check new ("http://localhost:3000");
+    string jsonFilePath = "./tests/resources/httpHeaderTest.json";
+    string readString = check io:fileReadString(jsonFilePath);
+    string replacedString = regex:replaceAll(readString, "(FUNC_NAME)", "httpHeaderTest9");
+    json readJson = check value:fromJsonString(replacedString);
+    json resp = check clientEndpoint->post("/post-hello-httpHeaderTest9", readJson);
+    json expectedResp = {"Outputs":{"resp":{"statusCode":400, "body":"no header value found for 'Hoste'", "headers":{"Content-Type":"text/plain"}}}, "Logs":[], "ReturnValue":null};
+    test:assertEquals(resp, expectedResp);
+}
+
+@test:Config {}
+function httpHeaderTest10() returns error? {
+    final http:Client clientEndpoint = check new ("http://localhost:3000");
+    string jsonFilePath = "./tests/resources/httpHeaderTest.json";
+    string readString = check io:fileReadString(jsonFilePath);
+    string replacedString = regex:replaceAll(readString, "(FUNC_NAME)", "httpHeaderTest10");
+    json readJson = check value:fromJsonString(replacedString);
+    json resp = check clientEndpoint->post("/post-hello-httpHeaderTest10", readJson);
+    json expectedResp = {"Outputs":{"resp":{"statusCode":400, "body":"no header value found for 'Hos'", "headers":{"Content-Type":"text/plain"}}}, "Logs":[], "ReturnValue":null};
+    test:assertEquals(resp, expectedResp);
+}
+
+@test:Config {}
+function httpHeaderTest11() returns error? {
```

Comment:
shall we make the test names, ballerina function names related descriptive? this is good for maintainability as if one test fails, we could get an idea easily but looking at the test results without diving into the code

Reply:
Sure.. I will change